### PR TITLE
[ENH] Interface new neuralforecast estimators

### PR DESCRIPTION
**Pull request by yarnabrina**
Interfaced GRU, DilatedRNN and TCN estimators from `neuralforecast`.

Note: these have been added in same pull request to avoid conflict due to multiple pull requests modifying same file simultanneously.

### CODE REVIEW

**Review by fkiraly — APPROVED** [review]
Hm, lots of parameters... but looks reasonable.

**Comment by yarnabrina** [comment]
Thanks.

FYI I get the merge option post approval of course, but since it's a new addition and the macos CI is not passing (getting cancelled due to timeout), I'll wait for approval or merge from other core developers instead of merging directly myself. It is frozen from my side.

**Comment by yarnabrina** [comment]
Well, merging myself since it has been a week from PR creation and approved 4 days ago. Hopefully that's okay.

**Pull request merged**
